sys/arm64/include/elf.h

```diff
@@ -57,12 +61,29 @@ typedef struct {	/* Auxiliary vector entry on initial stack */
 	long	a_type;			/* Entry type. */
 	union {
 		long	a_val;		/* Integer value. */
+#if __ELF_WORD_SIZE == 64 && !defined(__CHERI_PURE_CAPABILITY__)
```

Comment:
```suggestion
#if __SIZEOF_POINTER__ == 8
```

Reply:
These are not the same thing; e.g. CHERI64 would match the latter not the former. Leave it as it is; it’s also what we use elsewhere.

Reply:
Both `mips` and `riscv` use `__ELF_WORD_SIZE`

Reply:
Yeah I don't really care, I just prefer positive checks rather than negative ones.
`#if __ELF_WORD_SIZE == 64 && __SIZEOF_POINTER__ == 8` would also work for the cheri64 purecap case.

Reply:
In practice the sets of things that matches and the original code matches are the same, but only the original code says what it _means_. Your positive form makes you have to stop and think more IMO, and for anyone who doesn't know what CHERI is they'd be very confused, yet the mention of CHERI and capabilities in this version tells them what's actually going on.

Reply:
I was not sure whether __ELF_WORD_SIZE always matches __SIZEOF_POINTER, but it appears that e.g. the x32 ABI uses ELF32, so there are probably no ABIs other than CHERI where `__ELF_WORD_SIZE != __SIZEOF_POINTER__ * _CHAR_BIT`.

I really don't care, I just personally find __SIZEOF_POINTER__ easier to understand.